### PR TITLE
Add infores information for sankey for the automat sources

### PR DESCRIPTION
**Pull request by eKathleenCarter**
This code should pull the infores information from all automat-hosted sources. It makes a file called automat_infores_list.json with the information for the the Sankey.

### CODE REVIEW

**Comment by andrewsu** [comment]
The first entry in the generated file is

```
{
  "infores:agrkb": [
    "infores:agrkb"
  ],
```

should that be instead

```
{
  "infores:automat-agrkb": [
    "infores:agrkb"
  ],
```

and if yes, then presumably this would require an additional change to the `infores_catalog.yaml` file to add an entry for `infores:automat-agrkb`?

File: src/information_resource_registry/relation_map/Automat_infores.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    This a bit wonky and I suggest improvements, but this is how it runs now:
+    1) Get all the sources from smart api that start with "Automat"
+    2) Get the title of those sources: ie Automat-robokop(Trapi v1.5.0)
+    3) Extract just the knowledge provider name: ie robokop
+    4) Get the metadata of those sources from the automat endpoint: https://automat.renci.org/#/robokopkg
+        4a) If the knowledge provided name is not the same as the automat name, fix it
+    5) Make a dictionary  provenance: primary_knowledge_source
+
+    There should be a way to get the meta data directly from the smartAPI endpoint, but that requires getting
+    the smartAPI ID. There was no obious way to do this (to me).
```
Comment (andrewsu) [inline src/information_resource_registry/relation_map/Automat_infores.py:135]: obious -> obvious (flagged by the GH spelling checker)

File: src/information_resource_registry/relation_map/Automat_infores.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    automat_apis = [api['info']['title'] for api in all_apis]
+    KPs = extract_names(automat_apis)
+
+    infores_dict = fetch_metadata(KPs)
+
+    for k, v in infores_dict.items():
+        # Sort the list in alphabetical order
+        infores_dict[k].sort()
+
+    with open('src/information_resource_registry/relation_map/data/automat_infores_list.json', 'w') as outfile:
```
[the remark this answers is not in the document]
Comment (andrewsu) [reply src/information_resource_registry/relation_map/Automat_infores.py:151]: nevermind, I think the automation calls it from the root.  My bad!

[the remark this answers is not in the document]
Comment (eKathleenCarter) [reply src/information_resource_registry/relation_map/Automat_infores.py:151]: I think I just copied the path from the arax eample :-). But let me know if it needs to change

[the remark this answers is not in the document]
Comment (andrewsu) [reply src/information_resource_registry/relation_map/Automat_infores.py:151]: yup, you were right!

**Comment by EvanDietzMorris** [comment]
> The first entry in the generated file is
> 
> ```
> {
>   "infores:agrkb": [
>     "infores:agrkb"
>   ],
> ```
> 
> should that be instead
> 
> ```
> {
>   "infores:automat-agrkb": [
>     "infores:agrkb"
>   ],
> ```
> 
> and if yes, then presumably this would require an additional change to the `infores_catalog.yaml` file to add an entry for `infores:automat-agrkb`?

I haven't gotten around to looking at this more closely yet, but I think @andrewsu is probably right that we need to be using the automat infores ids for each automat (the aggregator infores), not the underlying data source. Those can be determined programmatically through the openapi spec for each Automat (openapi_spec["info"]["x-translator"]["infores"]). They should all be valid infores ids, in this specific case, there is an existing infores for this automat, it's just called something slightly different - infores:automat-genome-alliance.

**Comment by eKathleenCarter** [comment]
Addressed the following comment from @andrewsu using @EvanDietzMorris suggested programmatic way of getting the  automat infores ids (rather than the underlying sources) in commit 6894672

> The first entry in the generated file is
> 
> ```
> {
>   "infores:agrkb": [
>     "infores:agrkb"
>   ],
> ```
> 
> should that be instead
> 
> ```
> {
>   "infores:automat-agrkb": [
>     "infores:agrkb"
>   ],
> ```
> 
> and if yes, then presumably this would require an additional change to the `infores_catalog.yaml` file to add an entry for `infores:automat-agrkb`?

**Review by andrewsu — APPROVED** [review]
I'll leave it to @sierra-moxon to merge so she can keep an eye on the automation, but I've run the script locally and everything looks good to me!

**Pull request merged**